tests/test_aserver.py

```diff
@@ -26,6 +28,26 @@ def test_default_server(self):
         Aserver.shutdown_default_server()
         self.assertIsNone(s.stream)
 
+    def test_custom_channels(self):
+        ch = random.randint(1, 10)
```

Comment:
is it good practice to have test code depend on random arguments? wouldn't it be better to do the test on all possible channels? (assuming that the number is low / limited... Or test on 4 selected channels: first, in the middle, last, and beyond last (to test for invalid channel exception

Reply:
yeah, I agree that a precise testing strategy would be a better approach. If multiple scenarios with nearly the same setting should be evaluated, the test should be encapsulated and called multiple times with varying parameters (in this case: channels). Pytest fixtures could be useful (see [here](https://stackoverflow.com/a/41837463/1617563)).